### PR TITLE
upgrade: refine upgrade steps 

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/17138
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by joey-yez — APPROVED** [review]

File: tiflash-upgrade-guide.md [hunk 1]
```diff
@@ -23,11 +23,32 @@ To learn the standard upgrade process, see the following documents:
...
+
+    ```shell
+    tiup cluster stop <cluster-name> -R tiflash
+    ```
+
+2. Upgrade the TiDB cluster without restarting it (only updating the files):
+
+    ```shell
+    tiup cluster upgrade <cluster-name> <version> --offline`, such as `tiup cluster upgrade <cluster-name> v5.3.0 --offline
+    ```
```
Comment (hfxsd) [inline tiflash-upgrade-guide.md:45]:
```suggestion
    ```shell
    tiup cluster upgrade <cluster-name> <version> --offline 
    ```
    For example:

    ```shell
    tiup cluster upgrade <cluster-name> v5.3.0 --offline
    ```

File: tiflash-upgrade-guide.md [hunk 2]
```diff
@@ -23,11 +23,37 @@ To learn the standard upgrade process, see the following documents:
...
+    ```shell
+    tiup cluster stop <cluster-name> -R tiflash
+    ```
+
+2. Upgrade the TiDB cluster without restarting it (only updating the files):
+
+    ```shell
+    tiup cluster upgrade <cluster-name> <version> --offline 
+    ```
+    For example:
```
Comment (hfxsd) [inline tiflash-upgrade-guide.md:46]:
```suggestion

    For example:
```

File: tiflash-upgrade-guide.md [hunk 3]
```diff
@@ -23,11 +23,38 @@ To learn the standard upgrade process, see the following documents:
...
->     - Upgrade the TiDB cluster without restarting it (only updating the files): `tiup cluster upgrade <cluster-name> <version> --offline`, such as `tiup cluster upgrade <cluster-name> v5.3.0 --offline`
->     - Reload the TiDB cluster: `tiup cluster reload <cluster-name>`. After the reload, the TiFlash instance is started and you do not need to manually start it.
+
+## Upgrade TiFlash using TiUP
+
+To upgrade TiFlash from versions earlier than v5.3.0 to v5.3.0 or later, you must stop TiFlash and then upgrade it.
+
+If you are upgrading using TiUP, the TiUP version must be no later than v1.12.0. Otherwise, you cannot upgrade directly, but must first upgrade to an intermediate version.
+
+The following steps help you use TiUP to upgrade TiFlash without interrupting other components:
```
Comment (hfxsd) [inline tiflash-upgrade-guide.md:33]:
```suggestion
To upgrade TiFlash from versions earlier than v5.3.0 to v5.3.0 or later, you must stop TiFlash and then upgrade it. If you are upgrading using TiUP, note the following:

- If the TiUP cluster version is v1.12.0 or later, you cannot stop TiFlash and then upgrade it. If the target version requires a TiUP cluster version of v1.12.0 or later, it is recommended that you first use `tiup cluster:v1.11.3 <subcommamd>` to upgrade TiFlash to an intermediate version, perform an online upgrade of the TiDB cluster, upgrade the TiUP version, and then upgrade the TiDB cluster to the target version directly without stopping it.
- If the TiUP cluster version is earlier than v1.12.0, perform the following steps to upgrade TiFlash.

The following steps help you use TiUP to upgrade TiFlash without interrupting other components:
```

File: tiflash-upgrade-guide.md [hunk 4]
```diff
@@ -23,11 +23,39 @@ To learn the standard upgrade process, see the following documents:
...
 >
-> - To upgrade TiFlash from versions earlier than v5.3.0 to v5.3.0 or later, you should stop TiFlash and then upgrade it. The following steps help you upgrade TiFlash without interrupting other components:
->
->     - Stop the TiFlash instance: `tiup cluster stop <cluster-name> -R tiflash`
->     - Upgrade the TiDB cluster without restarting it (only updating the files): `tiup cluster upgrade <cluster-name> <version> --offline`, such as `tiup cluster upgrade <cluster-name> v5.3.0 --offline`
->     - Reload the TiDB cluster: `tiup cluster reload <cluster-name>`. After the reload, the TiFlash instance is started and you do not need to manually start it.
+
+## Upgrade TiFlash using TiUP
+
+To upgrade TiFlash from versions earlier than v5.3.0 to v5.3.0 or later, you must stop TiFlash and then upgrade it. If you are upgrading using TiUP, note the following:
```
Comment (lilin90) [inline tiflash-upgrade-guide.md:29]:
```suggestion
To upgrade TiFlash from versions earlier than v5.3.0 to v5.3.0 or later, you must stop TiFlash and then upgrade it. When you upgrade TiFlash using TiUP, note the following:
```

File: tiflash-upgrade-guide.md [hunk 5]
```diff
@@ -23,11 +23,39 @@ To learn the standard upgrade process, see the following documents:
...
+
+- If the TiUP cluster version is v1.12.0 or later, you cannot stop TiFlash and then upgrade it. If the target version requires a TiUP cluster version of v1.12.0 or later, it is recommended that you first use `tiup cluster:v1.11.3 <subcommamd>` to upgrade TiFlash to an intermediate version, perform an online upgrade of the TiDB cluster, upgrade the TiUP version, and then upgrade the TiDB cluster to the target version directly without stopping it.
```
Comment (lilin90) [inline tiflash-upgrade-guide.md:31]:
```suggestion
- If the TiUP cluster version is v1.12.0 or later, you cannot stop TiFlash and then upgrade it. If the target version requires a TiUP cluster version of v1.12.0 or later, it is recommended that you first use `tiup cluster:v1.11.3 <subcommand>` to upgrade TiFlash to an intermediate version, perform an online upgrade of the TiDB cluster, upgrade the TiUP version, and then upgrade the TiDB cluster to the target version directly without stopping it.
```

File: upgrade-tidb-using-tiup.md [hunk 6]
```diff
@@ -223,10 +223,7 @@ tiup cluster upgrade <cluster-name> v8.0.0
 >
 > + To keep a stable performance, make sure that all leaders in a TiKV instance are evicted before stopping the instance. You can set `--transfer-timeout` to a larger value, for example, `--transfer-timeout 3600` (unit: second).
 >
-> + To upgrade TiFlash from versions earlier than 5.3 to 5.3 or later, you should stop TiFlash and then upgrade it. The following steps help you upgrade TiFlash without interrupting other components:
->    1. Stop the TiFlash instance: `tiup cluster stop <cluster-name> -R tiflash`
->    2. Upgrade the TiDB cluster without restarting it (only updating the files): `tiup cluster upgrade <cluster-name> <version> --offline`, such as `tiup cluster upgrade <cluster-name> v6.3.0 --offline`
->    3. Reload the TiDB cluster: `tiup cluster reload <cluster-name>`. After the reload, the TiFlash instance is started and you do not need to manually start it.
+> + To upgrade TiFlash from versions earlier than 5.3 to 5.3 or later, you must stop TiFlash and then upgrade it, and the TiUP version must be no later than v1.12.0. For more information, see [Upgrade TiFlash using TiUP](/tiflash-upgrade-guide.md#upgrade-tiflash-using-tiup).
```
Comment (lilin90) [inline upgrade-tidb-using-tiup.md:226]:
```suggestion
> + To upgrade TiFlash from versions earlier than v5.3.0 to v5.3.0 or later, you must stop TiFlash and then upgrade it, and the TiUP version must be earlier than v1.12.0. For more information, see [Upgrade TiFlash using TiUP](/tiflash-upgrade-guide.md#upgrade-tiflash-using-tiup).
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**